control_plane/src/local_env.rs

```diff
@@ -102,6 +106,10 @@ pub struct OnDiskConfig {
     pub control_plane_api: Option<Url>,
     pub control_plane_compute_hook_api: Option<Url>,
     branch_name_mappings: HashMap<String, Vec<(TenantId, TimelineId)>>,
+    // Note: skip serializing because in compat tests old storage controller fails
+    // to load new config file. May be removed after this field is in release branch.
+    #[serde(skip_serializing_if = "std::ops::Not::not")]
```

Comment:
nit: shouldn't this just be `#[serde(skip)]` or something?

Reply:
No. The idea is "do not serialize if default". If it's explicitly turned on, we want to serialize it